sklearn/utils/_openmp_helpers.pyx

```diff
@@ -47,7 +47,10 @@ cpdef _openmp_effective_n_threads(n_threads=None):
         # to exceed the number of cpus.
         max_n_threads = omp_get_max_threads()
     else:
-        max_n_threads = min(omp_get_max_threads(), cpu_count())
+        max_n_threads = min(
+            omp_get_max_threads(),
+            cpu_count(only_physical_cores=only_physical_cores)
+        )
```

Comment:
This change impacts more scikit-learn in general and not just the test suite. I think we should document it in the changelog.

Reply:
I set the default to `False` so it has no impact on scikit-learn. We can discuss whether or not we want to change the default but I think it requires to be more careful and run some experiments (iirc some discussions in histgbt weren't clear about that)

Reply:
Indeed I had missed that. +1 for keeping the PR focused on the tests for now.

Reply:
We just found a case of this problem here: https://github.com/scikit-learn/scikit-learn/pull/25714#discussion_r1143767143.